crates/trigger/src/stdio.rs

```diff
@@ -36,52 +29,84 @@ impl Default for FollowComponents {
     }
 }
 
-/// Implements TriggerHooks, writing logs to a log file and (optionally) stderr
+/// Defines where to write logs
+pub enum LogDestination {
+    /// Write logs to stdout/stderr
+    Std,
+    /// Write logs to files in the directory
+    Dir(Option<PathBuf>),
+}
+
+/// Implements TriggerHooks, writing logs to a log file or stdout/stderr
 pub struct StdioLoggingTriggerHooks {
     follow_components: FollowComponents,
-    log_dir: Option<PathBuf>,
+    log: LogDestination,
 }
 
 impl StdioLoggingTriggerHooks {
-    pub fn new(follow_components: FollowComponents, log_dir: Option<PathBuf>) -> Self {
+    pub fn new(follow_components: FollowComponents, log: LogDestination) -> Self {
         Self {
             follow_components,
-            log_dir,
+            log,
         }
     }
 
-    fn component_stdio_writer(
+    fn component_log_writer(
         &self,
+        builder: &mut spin_core::StoreBuilder,
         component_id: &str,
-        log_suffix: &str,
-    ) -> Result<ComponentStdioWriter> {
-        let sanitized_component_id = sanitize_filename::sanitize(component_id);
-        let log_path = self
-            .log_dir
-            .as_deref()
-            .expect("log_dir should have been initialized in app_loaded")
-            .join(format!("{sanitized_component_id}_{log_suffix}.txt"));
-        let follow = self.follow_components.should_follow(component_id);
-        ComponentStdioWriter::new(&log_path, follow)
-            .with_context(|| format!("Failed to open log file {log_path:?}"))
+    ) -> anyhow::Result<()> {
+        if self.follow_components.should_follow(component_id) {
+            match &self.log {
+                LogDestination::Std => {
+                    builder.stdout_pipe(std::io::stdout());
+                    builder.stderr_pipe(std::io::stderr());
+                }
+                LogDestination::Dir(log_path) => {
+                    let log_path = log_path
+                        .as_deref()
+                        .expect("log should have been initialized in app_loaded");
```

Comment:
Seems like `LogDestination::Dir` should take a `PathBuf` rather than an `Option<PathBuf>` then, if the `None` case can never be valid?  (Again maybe I should be waiting for the refactoring.)

Reply:
This one preserves current behavior when `self.log` is `None` it gets assigned to default log directory inside `app_loaded` hook:
- https://github.com/fermyon/spin/blob/main/crates/trigger/src/stdio.rs#L73-L81
- https://github.com/fermyon/spin/blob/main/crates/trigger/src/stdio.rs#L59-L63

The only change here is that arg got updated from [`Option<T>` to `Option<Option<T>>`](https://docs.rs/clap/latest/clap/_derive/index.html#arg-types) to allow `-L/--log-dir` to be called w/o value (to use the default).

I would prefer to refactor it in the follow-up (if possible).